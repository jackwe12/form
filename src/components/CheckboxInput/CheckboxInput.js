import React from 'react';
// import ValidationAlert from '../ValidationAlert';

const CheckboxInput = ({ name, label, type, value, alert, setInputs }) => {
    return (
        <div>
            <label htmlFor={name} className="uk-form-label uk-margin-small-left">{label}</label>
            <input id={name} name={name} type={type} value={value||false} checked={value} onChange={setInputs} className={"uk-checkbox"} />
            {/* <ValidationAlert content={alert} /> */}
        </div>
    )
}

export default CheckboxInput;