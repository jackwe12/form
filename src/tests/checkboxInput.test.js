import CheckboxInput from '../components/CheckboxInput/CheckboxInput';
import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
configure({ adapter: new Adapter() });


const mockHandler = jest.fn();
const wrapper = shallow(
    <CheckboxInput 
        setInputs={mockHandler}
        {...{
        name: "emailOpt",
        label: "Email Opt Out",
        type: "checkbox",
        alert: "",
        }
    }

/>)
const wrapper2 = shallow(
    <CheckboxInput 
        setInputs={mockHandler}
        {...{
        name: "emailOpt",
        label: "Email Opt Out",
        type: "checkbox",
        alert: "error",
        }
    }

/>)

describe('Testing <CheckboxInput/>', ()=>{
    test('label tag renders correct text',()=>{
        expect(wrapper.find('label').text()).toEqual('Email Opt Out');
    });

    test('input tag renders correct prop',()=>{
        expect(wrapper.find('input').prop('name')).toEqual('emailOpt');
        expect(wrapper.find('input').prop('id')).toEqual('emailOpt');
        expect(wrapper.find('input').prop('value')).toEqual(false);

    });
    test('should call mock changeHandler when textarea changes',()=>{
        const input = wrapper.find('input');
        input.simulate('change');
        expect(mockHandler.mock.calls.length).toBe(1);
    });
});
