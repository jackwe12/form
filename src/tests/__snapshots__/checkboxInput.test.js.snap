// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <CheckboxInput/> snapshot testing 1`] = `
<div
  className="check-input-component"
>
  <label
    className="checkbox-label"
  >
    Email Opt Out
  </label>
  <input
    className="uk-checkbox"
    id="emailOpt"
    name="emailOpt"
    onChange={[MockFunction]}
    type="checkbox"
    value={false}
  />
</div>
`;
